Lib/test/test_complex.py

```diff
@@ -351,6 +354,45 @@ def test_pow_with_small_integer_exponents(self):
                     self.assertEqual(str(float_pow), str(int_pow))
                     self.assertEqual(str(complex_pow), str(int_pow))
 
+        # Check that complex numbers with special components
+        # are correctly handled.
+        values = [complex(*_) for _ in combinations([1, -1, 0.0, -0.0, 2,
```

Comment:
I'd also add the case `0` (integer type). In addition, it should be `combinations_with_replacement` rather, otherwise you will not generate all possible cases.